docs/en/collector.md

```diff
@@ -0,0 +1,152 @@
+# Collector
+
+## Collectors
+
+Yii Debug uses a concept named "collectors".
+Each collector decides what payload it needs to collect and exports the collected payload in order to save it into storage.
+
+A collector may work either both with HTTP requests and console runs, or individually.
+A collector may be either an event listener or a decorator to any service defined in the application DI container configuration.
+
+Take a look at the [`Yiisoft\Yii\Debug\Collector\CollectorInterface`](./src/Collector/CollectorInterface.php):
+
+```php
+namespace Yiisoft\Yii\Debug\Collector;
+
+interface CollectorInterface
+{
+    /**
+     * @return string Collector's name.
+     */
+    public function getName(): string;
+
+    /**
+     * Called once at application startup.
+     * Any initialization could be done here.
+     */
+    public function startup(): void;
+
+    /**
+     * Called once at application shutdown.
+     * Cleanup could be done here.
+     */
+    public function shutdown(): void;
+
+    /**
+     * @return array Data collected.
+     */
+    public function getCollected(): array;
+}
+```
+
+Use the trait to reduce the duplication of code and any possible bugs: [`\Yiisoft\Yii\Debug\Collector\CollectorTrait`](./src/Collector/CollectorTrait.php)
+
+All you need to create a collector is to implement the interface and register it in the configuration.
+
+### Example
+
+```php
+class MyCollector implements \Yiisoft\Yii\Debug\Collector\CollectorInterface
+{
+    use \Yiisoft\Yii\Debug\Collector\CollectorTrait;
+    
+    /**
+     * Payload collected during a run. 
+     */
+    private array $data = [];
+
+    public function getCollected() : array
+    {
+        return $this->data;
+    }
+}
+```
+
+When you implement collecting payload, it is also a good idea to implement data reset. With `CollectorTrait` it is as simple as adding `reset` method:
+```php
+    private function reset(): void
+    {
+        $this->data = [];
+    }
+```
+
+You can enable collector in application configuration as follows:
+
+```php
+return [
+    'yiisoft/yii-debug' => [
+        // if you want to register collector both for web requests and console runs
+        'collectors' => [
+            \App\Debug\AwesomeCollector::class,
+        ],
+        // if you want to register collector only for web requests
+        'collectors.web' => [
+            \App\Debug\AwesomeWebCollector::class,
+        ],
+        // if you want to register collector only for console runs
+        'collectors.console' => [
+            \App\Debug\AwesomeConsoleCollector::class,
+        ],
+    ],
+];
+```
+
+Under `yiisoft/yii-debug` configuration you may use:
+1. `collectors` key for both web and console runs
+2. `collectors.web` key only for web requests
+3. `collectors.console` key only for console runs
+
+> Do not register a collector for a session where the collector will not collect any payload.
+
+
+The lines above connect collectors with a debug extension run.
```

Comment:
Here we don't have a context of "debug extension run" yet.

Reply:
I don't remember what that means. I suppose "extension" = "package", this package.